packages/exhibitor/extra/dcos_zk_backup.py

```diff
@@ -0,0 +1,357 @@
+    shutil.rmtree(path=str(tmp_zookeeper_dir), ignore_errors=True)
+
+    print('Local ZooKeeper instance restored successfully')
+
+
+def _non_existing_file_path_existing_parent_dir(value: str) -> Path:
+    """
+    Validate that the value is a path to a file which does not exist
+    but the parent directory tree exists.
+    """
+    path = Path(value)
+    if path.exists():
+        raise ArgumentTypeError('{} already exists'.format(path))
+    if not Path(path.parent).exists():
+        raise ArgumentTypeError(
+            '{} parent directory does not exist'.format(path),
+        )
+    return path.absolute()
+
+
+def _existing_file_path(value: str) -> Path:
+    """
+    Validate that the value is a file which does exist on the file system.
+    """
+    path = Path(value)
+    if not path.exists():
+        raise ArgumentTypeError('{} does not exist'.format(path))
+    if not path.is_file():
+        raise ArgumentTypeError('{} is not a file'.format(path))
+    return path.absolute()
+
+
+def _existing_dir_path(value: str) -> Path:
+    """
+    Validate that the value is a directory which does exist on the file system.
+    """
+    path = Path(value)
+    if not path.exists():
+        raise ArgumentTypeError('{} does not exist'.format(path))
+    if not path.is_dir():
+        raise ArgumentTypeError('{} is not a directory'.format(path))
+    return path.absolute()
+
+
+class DCOSZooKeeperCli:
+    """
+    Minimal CLI to backup/restore the local DC/OS ZooKeeper instance.
+    """
+
+    def __init__(self) -> None:
+        """
+        Present CLI command choices.
+        """
+        parser = ArgumentParser(
+            description=(
+                'Command line utility to more safely interact with the local '
+                'ZooKeeper instance on DC/OS master nodes'
+            )
+        )
+        parser.add_argument(
+            'command',
+            type=str,
+            choices=[
+                'backup',
+                'restore',
+            ],
+            help='CLI commands available',
+        )
+        args = parser.parse_args(sys.argv[1:2])
+        getattr(self, args.command)()
+
+    def backup(self) -> None:
+        """
+        Procedure invoked on `backup` command.
+        """
+        parser = ArgumentParser(
+            description=(
+                'Create a backup of the ZooKeeper instance running on this '
+                'DC/OS master node'
+            )
+        )
+        parser.add_argument(
+            'backup_path',
+            type=_non_existing_file_path_existing_parent_dir,
+            help=(
+                'File path that the gzipped ZooKeeper backup tar archive will '
+                'be written to'
+            ),
+        )
+        parser.add_argument(
+            '-t', '--tmp-dir',
+            type=_existing_dir_path,
+            help=(
+                'Location of an existing directory to be used as temporary '
+                'directory in the process'
```

Comment:
What's temporary about this? Will this script work if I do not provide this? What's the default?

Reply:
maybe say something like "use this if the default does not work".